tests/apps/pubsub-subscriber/app.go

```diff
@@ -91,6 +98,19 @@ func configureSubscribeHandler(w http.ResponseWriter, r *http.Request) {
 func subscribeHandler(w http.ResponseWriter, r *http.Request) {
 	log.Printf("aHandler is called %s\n", r.URL)
 
+	if respondWithRetry {
+		// do not store received messages, respond with success but a retry status
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: "retry later",
+			Status:  "RETRY",
+		})
+		return
+	} else if respondWithError {
+		// do not store received messages, respond with error
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Any non-2xx status other than 404 will be retried. 
https://github.com/dapr/dapr/blob/master/pkg/runtime/runtime.go#L1126